### ISSUE TITLE
[FEAT] Improve accessibility throughout the UI

### ISSUE THREAD

**Issue by dikunong** [Team-5-Cyan-Camel/Flatmates#83]
The user interface has many components with accessibility issues - mostly to do with keyboard navigation (needed by the motor-impaired and the visually-impaired/blind). Some of the issues I noticed:

* The top-level `Navbar` has `<Link>` elements that contain `<a>` elements without a `href` attribute, generating links that do nothing when clicked (referenced in #82)
* The <Cross> element, wherever it is used, is completely unselectable via keyboard
* The Roster `Navbar` contains items with insufficient contrast ratios (light grey on dark grey text hard to read)

Accessibility is a requirement of any user-facing software interface. Read more about designing and implementing accessible UIs [here](https://developer.mozilla.org/en-US/docs/Learn/Accessibility) and [here](https://developers.google.com/web/fundamentals/accessibility). It can get pretty involved, but two easy principles are "can I use this application without a mouse?" and "do my colours have enough contrast?"

**Comment by JChamberlain28** [comment]
These are really good points. A good enhancement to add.

**Comment by JChamberlain28** [comment]
We were balancing many priorities so this small feature was not included in this release.
